### DISCUSSION TITLE
Can't create context menus for certain commands in tree view

### DISCUSSION THREAD

**Discussion by david-tejada** [microsoft/vscode-discussions#947] [Extension Development QnA] [answered]
I want to reproduce the native explorer context menu for an accessibility extension. I am being unable to create context menus for certain commands. For example, in the following example the context menus for `revealFileInOS` and `openToSide` aren't being created while the rest are:

```json
"menus": {
            "view/item/context": [
                {
                    "command": "fileutils.removeFile",
                    "when": "view == filetree"
                },
                {
                    "command": "revealFileInOS",
                    "when": "!editorFocus && view == filetree"
                },
                {
                    "command": "git.timeline.selectForCompare",
                    "when": "!editorFocus && view == filetree"
                },
                {
                    "command": "fileutils.newFolderAtRoot",
                    "when": "!editorFocus && view == filetree"
                },
                {
                    "command": "fileutils.copyFileName",
                    "when": "!editorFocus && view == filetree"
                },
                {
                    "command": "openToSide",
                    "when": "!editorFocus && view == filetree"
                }
            ]
        }
```

Is there any way to create context menus for those commands?

**Comment by olguzzar** [comment]
Hi @david-tejada, bumping this as it's been a while without any activity. Are you still facing this problem?

**Reply by david-tejada** [reply]
Yes, thanks. I'd still like to find a solution for this.

**Comment by isidorn** [comment]
This should just work. I think you are using the when context wrongly. So I suggest to check out some other contextKeys, or your extension can contribute a contextKey and control it for full control.

**Reply by david-tejada** [reply]
That's not the issue. I just realized there is an error message: "Menu item references a command `revealFileInOS` which is not defined in the 'commands' section.". It seems it's interpreting that command as being part of the extension. Is there a way to use the provided `revealFileInOS` command or do I have to create a command that uses that internally?

**Comment by burkeholland** [comment, marked as answer]
I don't think you can directly call the `revealFileInOS` command like this because you need to construct the File URI. I was able to get this to work by wrapping in a command and handling the uri scheme....

https://github.com/burkeholland/menu-tester

**Reply by david-tejada** [reply]
Yes, I think this is the right answer. Thanks for taking the time to look into this.
